CRM/Core/Controller.php

```diff
@@ -457,8 +457,8 @@ public function addPages(&$stateMachine, $action = CRM_Core_Action::NONE) {
       if ($options) {
         $$stateName->setOptions($options);
       }
-      if (property_exists($$stateName, 'urlPath') && isset($_GET[CRM_Core_Config::singleton()->userFrameworkURLVar])) {
-        $$stateName->urlPath = explode('/', $_GET[CRM_Core_Config::singleton()->userFrameworkURLVar]);
+      if (property_exists($$stateName, 'urlPath')) {
+        $$stateName->urlPath = explode('/', (string) CRM_Utils_System::getUrlPath());
```

Comment:
So, here we are setting CRM_Core_StateMachine->urlPath to the 'q' value (for drupal) or NULL if there is not one 

(question - could the property exist and be set & be reset to NULL by this?)